aptos-move/move-examples/bonding_curve_launchpad/sources/bonding_curve_launchpad.move

```diff
@@ -0,0 +1,238 @@
+    use resource_account::resource_signer_holder;
+
+    const INITIAL_NEW_FA_RESERVE_u64: u64 = 8_003_000_000_000;
+    const INITIAL_NEW_FA_RESERVE: u128 = 8_003_000_000_000;
+
+    const EFA_EXISTS_ALREADY: u64 = 10;
+    const EFA_DOES_NOT_EXIST: u64 = 11;
+    const EFA_FROZEN: u64 = 13;
+    const ELIQUIDITY_PAIR_SWAP_AMOUNTIN_INVALID: u64 = 110;
+
+    //---------------------------Events---------------------------
+    #[event]
+    struct FungibleAssetCreated has store, drop {
+        name: string::String,
+        symbol: string::String,
+        max_supply: u128,
+        decimals: u8,
+        icon_uri: string::String,
+        project_uri: string::String
+    }
+
+    //---------------------------Structs---------------------------
+    struct LaunchPad has key {
+        key_to_fa_data: SmartTable<FAKey, FAController>,
+    }
+    struct FAKey has store, copy, drop {
+        name: string::String,
+        symbol: string::String,
+    }
+    struct FAController has key, store {
+        mint_ref: fungible_asset::MintRef,
+        burn_ref: fungible_asset::BurnRef,
+        transfer_ref: fungible_asset::TransferRef
+    }
+
+    //---------------------------Init---------------------------
+    fun init_module(account: &signer) {
+        let fa_smartTable: LaunchPad = LaunchPad {
+            key_to_fa_data: smart_table::new()
+        };
+        move_to(account, fa_smartTable);
+    }
+
+    //---------------------------Dispatchable Standard---------------------------
+    //* FA is restricted in transfers, until an APT reserves threshold is met.
+    //* - Since transfer_ref is only available to our permissioned actions, this can't be used by bad actors.
+    public fun withdraw<T: key>(
+        store: Object<T>,
+        amount: u64,
+        transfer_ref: &fungible_asset::TransferRef
+    ): FungibleAsset {
+        let metadata = fungible_asset::transfer_ref_metadata(transfer_ref);
+        assert!(!liquidity_pair::get_is_frozen_metadata(metadata), EFA_FROZEN);
+        fungible_asset::withdraw_with_ref(transfer_ref, store, amount)
+    }
+
+    //---------------------------Bonding Curve Launchpad (BCL)---------------------------
+    // * Creates new FA and store FA owner obj on launchpad.
+    public entry fun create_fa_pair(
+        account: &signer,
+        apt_initialPurchaseAmountIn: u64,
+        name: string::String,
+        symbol: string::String,
+        max_supply: u128,
+        decimals: u8,
+        icon_uri: string::String,
+        project_uri: string::String
+    ) acquires LaunchPad {
+        let fa_key = FAKey { name, symbol };
+        let fa_address = create_fa(fa_key, name, symbol, max_supply, decimals, icon_uri, project_uri);
+        let fa_metadata_obj: Object<Metadata> = object::address_to_object(fa_address);
+        let fa_smartTable = borrow_global_mut<LaunchPad>(@resource_account);
+        let transfer_ref = &smart_table::borrow(&mut fa_smartTable.key_to_fa_data, fa_key).transfer_ref;
```

Comment:
&mut fa_smartTable.key_to_fa_data is not necessary. You just need an immutable reference